xml/src/main/java/org/ehcache/xml/BaseConfigParser.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright Terracotta, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.ehcache.xml;
+
+import org.ehcache.xml.exceptions.XmlConfigurationException;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.xml.sax.SAXException;
+
+import java.io.IOException;
+import java.util.Objects;
+
+import javax.xml.parsers.ParserConfigurationException;
+import javax.xml.transform.Source;
+
+/**
+ * BaseConfigParser
+ */
+public abstract class BaseConfigParser<T> {
+  private final Class<T> typeParameterClass;
+
+  public BaseConfigParser(Class<T> typeParameterClass) {
+    this.typeParameterClass = typeParameterClass;
```

Comment:
If you want to have fun, you can actually guess this parameter with `getClass().getGenericSupertype.getParameters()[0]`